pkg/proc/bininfo.go

```diff
@@ -1656,14 +1655,17 @@ func loadBinaryInfoPE(bi *BinaryInfo, image *Image, path string, entryPoint uint
 		return err
 	}
 
-	//TODO(aarzilli): actually test this when Go supports PIE buildmode on Windows.
-	opth := peFile.OptionalHeader.(*pe.OptionalHeader64)
-	if entryPoint != 0 {
-		image.StaticBase = entryPoint - opth.ImageBase
-	} else {
-		if opth.DllCharacteristics&_IMAGE_DLLCHARACTERISTICS_DYNAMIC_BASE != 0 {
```

Comment:
Why did you remove this check?

Reply:
windows/arm64 binaries are always PIE, even if built with `-buildmode=exe`, so we need to support `entryPoint==0`. I don't know why this check was added, but as per the TODO comment I supposed it was just that it still wasn't tested with a real program, which I'm doing now.

Reply:
The TODO command was obsolete, we did end up checking that this worked on windows. AFAIK PIE binaries always have non-zero entryPoint, the check is to make sure that we don't proceed with a PIE binary for which we weren't able to retrieve the entryPoint.

Reply:
There are some tests calling `LoadBinaryInfo` with 0 as hardcoded entry point, namely `TestDwarfVersion` and `TestRegabiFlagSentinel`. These tests fail on windows/arm64 because the binary is PIE and the entry point is 0.

Anyway, I can revert this change and fix the tests by using a fake entry point, as they don't really need this information.

Reply:
Yes, it's better to change the entry point used by the tests.